bootstrap/sponge/src/main/java/org/geysermc/platform/sponge/command/GeyserSpongeCommandExecutor.java

```diff
@@ -25,48 +25,49 @@
 
 package org.geysermc.platform.sponge.command;
 
-import org.geysermc.connector.GeyserConnector;
+import net.kyori.adventure.text.Component;
 import org.geysermc.connector.command.CommandExecutor;
+import org.geysermc.connector.command.CommandManager;
 import org.geysermc.connector.command.CommandSender;
 import org.geysermc.connector.command.GeyserCommand;
 import org.geysermc.connector.common.ChatColor;
 import org.geysermc.connector.network.session.GeyserSession;
 import org.geysermc.connector.utils.LanguageUtils;
-import org.spongepowered.api.command.CommandCallable;
+import org.jetbrains.annotations.NotNull;
+import org.spongepowered.api.command.Command;
+import org.spongepowered.api.command.CommandCause;
+import org.spongepowered.api.command.CommandCompletion;
 import org.spongepowered.api.command.CommandResult;
-import org.spongepowered.api.command.CommandSource;
-import org.spongepowered.api.text.Text;
-import org.spongepowered.api.world.Location;
-import org.spongepowered.api.world.World;
+import org.spongepowered.api.command.parameter.ArgumentReader;
 
-import javax.annotation.Nullable;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
-public class GeyserSpongeCommandExecutor extends CommandExecutor implements CommandCallable {
+public class GeyserSpongeCommandExecutor extends CommandExecutor implements Command.Raw {
 
-    public GeyserSpongeCommandExecutor(GeyserConnector connector) {
-        super(connector);
+    public GeyserSpongeCommandExecutor(CommandManager commandManager) {
+        super(commandManager);
     }
 
     @Override
-    public CommandResult process(CommandSource source, String arguments) {
-        CommandSender commandSender = new SpongeCommandSender(source);
-        GeyserSession session = getGeyserSession(commandSender);
+    public CommandResult process(CommandCause cause, ArgumentReader.Mutable arguments) {
+        CommandSender commandSender = new SpongeCommandSender(cause);
+        GeyserSession session = commandSender.getGeyserSession();
 
-        String[] args = arguments.split(" ");
+        String[] args = arguments.input().split(" "); //todo: this probably doesn't work
         if (args.length > 0) {
             GeyserCommand command = getCommand(args[0]);
             if (command != null) {
-                if (!source.hasPermission(command.getPermission())) {
+                if (!cause.hasPermission(command.getPermission())) {
                     // Not ideal to use log here but we dont get a session
-                    source.sendMessage(Text.of(ChatColor.RED + LanguageUtils.getLocaleStringLog("geyser.bootstrap.command.permission_fail")));
+                    cause.audience().sendMessage(Component.text(ChatColor.RED + LanguageUtils.getLocaleStringLog("geyser.bootstrap.command.permission_fail")));
```

Comment:
Instead of invoking ChatColor.RED here, we should probably use `.color(NamedTextColor.RED)`. See https://docs.adventure.kyori.net/text#creating-components as an example.